CONTRIBUTING.md

```diff
@@ -1,25 +1,160 @@
+Before you begin, ensure you have the following installed:
+
+### Project Overview
+
+Our codebase is organized as follows:
+
+- `.github/`: Contains GitHub files like issue templates and workflows
+- `recipes/`: Contains all the codemods
+- `utils/`: An npm workspace with utility functions used by the codemods (including [ast-grep](https://ast-grep.github.io/) utilities)
+
+## Codemod Development
+
+### Structure of a Codemod
+
+Each codemod resides in its own directory under `recipes/` and should include:
+
+| File | Purpose |
+|------|---------|
+| `README.md` | Description, purpose, and usage instructions |
+| `package.json` | Package manifest |
+| `src/workflow.ts` | Main entry point using the `jssg` codemod API |
+| `codemod.yml` | Codemod manifest file |
+| `workflow.yml` | Workflow definition file |
+| `tests/` | Test suite using `jssg` testing utilities |
+| `tsconfig.json` | TypeScript configuration |
+### Example Files
+**`src/workflow.ts` example:**
+```ts
+import type { SgRoot, Edit } from "@codemod.com/jssg-types/main";
+/**
+ * Transform function that converts deprecated api.fn calls
+ * to the new api.fn syntax.
+ *
+ * Handles:
+ * 1. api.fn(<args>)
+ * 2. api.fn(<args>, { recursive: true })
+ * ...
+ */
+export default function transform(root: SgRoot): string | null {
+	const rootNode = root.root();
+	let hasChanges = false;
+	const edits: Edit[] = [];
+	// do some transformation
+	if (!hasChanges) return null;
+	return rootNode.commitEdits(edits);
+}
+```
+**`codemod.yml` example:**
+```yaml
+schema_version: "1.0"
+name: "@nodejs/<codemod-name>"
+version: 0.0.1
+description: <Your codemod description>
+author: <Your Name>
+license: MIT
+workflow: workflow.yaml
+category: migration
+targets:
+  languages:
+    - javascript
+    - typescript
+keywords:
+  - transformation
+  - migration
+registry:
+  access: public
+  visibility: public
+```
+
+## Useful Resources
+
+- [Codemod CLI Reference](https://docs.codemod.com/cli/cli-reference)
+- [Codemod Workflow Documentation](https://docs.codemod.com/cli/workflows)
+- [Codemod Studio Documentation](https://docs.codemod.com/codemod-studio)
+- [JSSG API Reference](https://docs.codemod.com/cli/cli-reference#cli-command-reference)
+- [AST-grep Documentation](https://ast-grep.github.io/)
 
 ## Before pushing a commit
 
 A convenient superset of checks is available via `node --run pre-commit`, which automatically fixes formatting and linting issues (that are safe to fix), checks types, and runs tests. Changes resulting from this should be committed.
 
 > [!WARNING]
 > Some integration tests modify fixtures because they run the entire codemod. Remember to use the `git restore` command to restore these files before pushing a commit.
+
+## Development Workflow
+
+### Before Pushing a Commit
+
+Run our comprehensive check suite:
+
+```bash
+node --run pre-commit
+```
```

Comment:
This is mentioned a few lines above. I think they should be de-duplicated (no opinion on which to keep).

Reply:
simplify